openedx/core/djangoapps/schedules/tasks.py

```diff
@@ -85,6 +86,45 @@ def _recurring_nudge_schedule_send(site_id, msg_str):
 
 
 def _recurring_nudge_schedules_for_hour(target_hour, org_list, exclude_orgs=False):
+
+    users, schedules = _gather_users_and_schedules_for_target_hour(target_hour, org_list, exclude_orgs)
+
+    dashboard_relative_url = reverse('dashboard')
+    for (user, user_schedules) in groupby(schedules, lambda s: s.enrollment.user):
+        user_schedules = list(user_schedules)
+        course_id_strs = [str(schedule.enrollment.course_id) for schedule in user_schedules]
+
+        def absolute_url(relative_path):
+            return u'{}{}'.format(settings.LMS_ROOT_URL, urlquote(relative_path))
+
+        first_schedule = user_schedules[0]
+        upgrade_data = check_and_get_upgrade_link(user, first_schedule.enrollment.course_id)
+
+        template_context = {
+            'student_name': user.profile.name,
+
+            'course_name': first_schedule.enrollment.course.display_name,
+            'course_url': absolute_url(reverse('course_root', args=[str(first_schedule.enrollment.course_id)])),
+
+            # This is used by the bulk email optout policy
+            'course_ids': course_id_strs,
+            # Platform information
+            'homepage_url': encode_url(marketing_link('ROOT')),
+            'dashboard_url': absolute_url(dashboard_relative_url),
+            'template_revision': settings.EDX_PLATFORM_REVISION,
+            'platform_name': settings.PLATFORM_NAME,
+            'contact_mailing_address': settings.CONTACT_MAILING_ADDRESS,
+            'social_media_urls': encode_urls_in_dict(getattr(settings, 'SOCIAL_MEDIA_FOOTER_URLS', {})),
+            'mobile_store_urls': encode_urls_in_dict(getattr(settings, 'MOBILE_STORE_URLS', {})),
+        }
+
+        # Information for including upsell messaging in template.
+        add_upsell_button_to_email_template(upgrade_data, template_context)
```

Comment:
nit: convention is to prefix "private" functions like this with a single underscore "_"